Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -216,16 +218,22 @@ public async Task PutObjectAsync(string bucketName, string objectName, Stream da
         {
             utils.ValidateBucketName(bucketName);
             utils.ValidateObjectName(objectName);
+
             var sseHeaders = new Dictionary<string, string>();
+            var meta = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            if (metaData != null) {
+                foreach (KeyValuePair<string, string> p in metaData)
+                {
+                    var key = p.Key;
+                    if (!supportedHeaders.Contains(p.Key) && !p.Key.ToLower().StartsWith("x-amz-meta-"))
```

Comment:
Can we write a separate function to check this? As we are using it in more than one place.